server/src/main/java/org/apache/druid/server/http/DataSegmentDto.java

```diff
@@ -0,0 +1,224 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.http;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.apache.druid.guice.annotations.PublicApi;
+import org.apache.druid.jackson.CommaListJoinDeserializer;
+import org.apache.druid.jackson.CommaListJoinSerializer;
+import org.apache.druid.timeline.CompactionState;
+import org.apache.druid.timeline.DataSegment;
+import org.apache.druid.timeline.SegmentId;
+import org.apache.druid.timeline.partition.ShardSpec;
+import org.joda.time.DateTime;
+import org.joda.time.Interval;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@link DataSegment} to be transmitted externally
+ */
+@PublicApi
+public class DataSegmentDto
+{
+  private final DataSegment dataSegment;
+  private final DateTime createdDate;
+  @Nullable
+  private final DateTime usedStatusLastUpdatedDate;
+
+  public DataSegmentDto(
+      DataSegment dataSegment,
+      DateTime createdDate,
+      @Nullable DateTime usedStatusLastUpdatedDate
+  )
+  {
+    this.dataSegment = dataSegment;
+    this.createdDate = createdDate;
+    this.usedStatusLastUpdatedDate = usedStatusLastUpdatedDate;
+  }
+
+  @JsonCreator
+  public DataSegmentDto(
+      @JsonProperty("dataSource") String dataSource,
+      @JsonProperty("interval") Interval interval,
+      @JsonProperty("version") String version,
+      // use `Map` *NOT* `LoadSpec` because we want to do lazy materialization to prevent dependency pollution
+      @JsonProperty("loadSpec") @Nullable Map<String, Object> loadSpec,
+      @JsonProperty("dimensions")
+      @JsonDeserialize(using = CommaListJoinDeserializer.class)
+      @Nullable
+      List<String> dimensions,
+      @JsonProperty("metrics")
+      @JsonDeserialize(using = CommaListJoinDeserializer.class)
+      @Nullable
+      List<String> metrics,
+      @JsonProperty("shardSpec") @Nullable ShardSpec shardSpec,
+      @JsonProperty("lastCompactionState") @Nullable CompactionState lastCompactionState,
+      @JsonProperty("binaryVersion") Integer binaryVersion,
+      @JsonProperty("size") long size,
```

Comment:
Hmm, do we need to duplicate all the fields from the `DataSegment` class here? If the jackson annotation wireup is done correctly, I think we can get rid of this constructor altogether and just have the above 3-arg constructor with `@JsonCreator`

Reply:
I was trying to do this so that we dont break the existing api that returned `DataSegment` object previously. I believe if we remove these annotations, the object will be serialized with the DataSegment fields within an objecr rather than flattened as they were before, which I think would be a breaking change.

Reply:
Since these changes are [internal](https://github.com/apache/druid/pull/15721) and are only used by the unused segments API, which hasn't been released yet, there's no need to maintain backwards compatibility

Reply:
fixed